### ISSUE TITLE
Unique identifier

### ISSUE THREAD

**Issue by jracusin** [nasa-gcn/gcn-schema#54]
In order to be able to cite notice messages individually, we need to add a unique identifier for each message in the producer class.  This could include a unique string for each producer than a alphanumeric string for each message.

**Comment by dakota002** [comment]
@Tohuvavohu a quick update on this: I made a PR for the gcn.nasa.gov that updates the sample code to print out the offset number. 

If you want to print it in a consumer, you can add  `print(f'{message.topic()}: #{message.offset()}')` to the consuming loop. 
You can use the offset and topic to directly reference specific Notices. Here is an example of retrieving a gcn.classic.text.SWIFT_ACTUAL_POINTDIR notice using the specific offset number: 33893 (I got this number yesterday using the message.offset() example). The python gcn-kafka library is a wrapper around confluent_kafka, so you should already have the package installed


```
from gcn_kafka import Consumer
from confluent_kafka import cimpl

# Connect as a consumer.
# Warning: don't share the client secret with others.
consumer = Consumer(client_id='your-client-id',
                    client_secret='your-client-secret')


topic = "gcn.classic.text.SWIFT_ACTUAL_POINTDIR"
pt = cimpl.TopicPartition(topic, 0, 33893)
consumer.commit(offsets=[pt])
consumer.subscribe([topic])

for message in consumer.consume(num_messages=1):
    value = message.value()
    print(f'{message.topic()}: {message.offset()}')
    print(value)
```

**Comment by Tohuvavohu** [comment]
Thanks @dakota002 ! :) 
Definitely should be in the documentation for users (i see that in your PR now)

**Comment by Tohuvavohu** [comment]
Do you think it makes sense to append the ID number to the alert packet itself?
Then people who save the alert.json on receipt will be able to reference the ID.
Don't know if this is feasible givne how the offset number is assigned.
Thoughts?

**Comment by dakota002** [comment]
I would say it technically is. If you consider the alert packet as the whole message object, and keep in mind that the JSON is just the `value` . I definitely agree though that this info should be more visible in our documentation. Here is some more information on the [Message class](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Message)

**Comment by Tohuvavohu** [comment]
Right, but can it be appended to the `value` object itself? I imagine many will want to save the value of the alert to a json file, and being able to consistently reference and find the ID would be very useful for users, I think.

**Comment by lpsinger** [comment]
You don't really know the offset until the Kafka broker has ingested the record. So that's not really possible.

**Comment by Tohuvavohu** [comment]
figured that might be the case, thanks.

**Comment by blaufuss** [comment]
Just to clarify: this is what's intended to be included when one uses the "reference"  the gcn-schema/core/FollowUp schema, like:

  "reference": { "gcn.notices.LVK.alert": 6666 },

Additionally, confirming there's no ability to access this if you're using older GCN interfaces, correct?
